File: tone_analysis_service/crud.py
from database import database
from bson import ObjectId

question_collection = database["questions"]
interview_template_collection = database["interview_templates"]
user_collection = database["users"]
transcription_collection = database['transcriptions']
score_collection = database['scores']
audio_collection = database['audio_records']

# Function to create a transcription record
async def create_audio_record(record: dict):
    result = await transcription_collection.insert_one(record)
    return str(result.inserted_id)

# Create a new score record
async def create_score_record(record: dict):
    result = await score_collection.insert_one(record)
    return str(result.inserted_id)

# Get question record by id
async def get_question_record(question_id: str):
    result = await question_collection.find_one({"_id": ObjectId(question_id)})
    return result

# get interview template record by id
async def get_interview_template_record(interview_template_id: str):
    result = await interview_template_collection.find_one({"_id": ObjectId(interview_template_id)})
    return result

# Get user by id
async def get_user_record(user_id: str):
    user = await user_collection.find_one({"_id": ObjectId(user_id)})
    return user

async def create_tone_record(record: dict) -> str:
    result = await audio_collection.insert_one(record)
    return str(result.inserted_id)

File: crud_api_service/crud.js
const transcriptions = require('./models/transcriptions');
const images = require('./models/images');
const scores = require('./models/scores');
const questions = require('./models/questions');
const interviews = require('./models/interviews');
const users = require('./models/users');
const interview_templates = require('./models/interview_template');


// Get an image record by ID
async function getImageRecord(id) {
    const record = await images.findOne({ _id: id });
    return record;
}

// Get all image records
async function getAllImageRecords() {
    const records = await images.find();
    return records;
}

// Create question record
async function createQuestionRecord(questionData) {
    const question = new questions(questionData);
    const result = await question.save();
    return result._id.toString();
}

// Get question record by ID
async function getQuestionRecord(questionId) {
    const question = await questions.findOne({ _id: questionId });
    return question;
}

// Get all questions
async function getAllQuestions() {
    const questions = await questions.find();
    return questions;
}

// Create interview template record
async function createInterviewDB(interviewData) {
    const interview = new interview_template(interviewData);
    const result = await interview.save();
    return result._id.toString();
}

// Get interview template record by ID
async function getInterviewTemplateRecord(interviewTemplateId) {
    const interviewTemplate = await interview_templates.findOne({ _id: interviewTemplateId });
    return interviewTemplate;
}

// Get all interview templates
async function getAllInterviews() {
    const interviews = await interview_templates.find();
    return interviews;
}

// Get user record by ID
async function getUserRecord(userId) {
    const user = await users.findOne({ _id: userId });
    return user;
}

// Get all scores
async function getScores() {
    try {
        const response = await images.find().populate('question_id').populate('interview_id').populate('user_id');
        return response;
    } catch (error) {
        console.error('Error fetching scores:', error);
    }
}

module.exports = {
    getImageRecord,
    getAllImageRecords,
    createQuestionRecord,
    getQuestionRecord,
    getAllQuestions,
    createInterviewDB,
    getInterviewTemplateRecord,
    getAllInterviews,
    getUserRecord,
    getScores
};
